wsproto/connection.py

```diff
@@ -275,6 +275,11 @@ def events(self):
                 if frame.opcode is Opcode.PING:
                     assert frame.frame_finished and frame.message_finished
                     self._outgoing += self._proto.pong(frame.payload)
+                    yield PingReceived(frame.payload)
+
+                elif frame.opcode is Opcode.PONG:
+                    assert frame.frame_finished and frame.message_finished
+                    yield PongReceived(frame.payload)
```

Comment:
If the protocol lets either side send PONGs arbitrarily, how do we know that a particular PONG was a response to a PING that we sent?